operator/scheduler/client.go

```diff
@@ -108,14 +113,54 @@ func (s *SchedulerClient) startEventHanders(namespace string, conn *grpc.ClientC
 	}()
 	go func() {
 		for {
-			err := retryFn(s.SubscribeExperimentEvents, conn, namespace, s.logger)
+			err := retryFn(s.SubscribeExperimentEvents, conn, namespace, s.logger.WithName("SubscribeExperimentEvents"))
 			if err != nil {
 				s.logger.Error(err, "Subscribe ended for experiment events", "namespace", namespace)
 			} else {
 				s.logger.Info("Subscribe ended for experiment events", "namespace", namespace)
 			}
 		}
 	}()
+	go func() {
+		for {
+			err := retryFn(s.SubscribeControlPlaneEvents, conn, namespace, s.logger.WithName("SubscribeControlPlaneEvents"))
+			if err != nil {
+				s.logger.Error(err, "Subscribe ended for control plane events", "namespace", namespace)
+			} else {
+				s.logger.Info("Subscribe ended for control plane events", "namespace", namespace)
+			}
+		}
+	}()
+}
+
+func (s *SchedulerClient) handleStateOnReconnect(context context.Context, grpcClient scheduler.SchedulerClient, namespace string) error {
+	// on new reconnects we send a list of servers to the schedule
+	err := s.handleRegisteredServers(context, grpcClient, namespace)
```

Comment:
For a future PR: all those contexts end up as context.Background(), so not cancellable.